core/sdk-core/src/main/java/software/amazon/awssdk/core/compression/CompressionType.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.compression;
+
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.Set;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.core.internal.compression.GzipCompressor;
+import software.amazon.awssdk.utils.internal.EnumUtils;
+
+/**
+ * The supported compression algorithms for operations with the requestCompression trait. Each supported algorithm will have an
+ * {@link Compressor} implementation.
+ */
+@SdkInternalApi
+public enum CompressionType {
+
+    GZIP("gzip"),
+
+    UNKNOWN_TO_SDK_VERSION(null);
+
+    private static final Map<String, CompressionType> VALUE_MAP = EnumUtils.uniqueIndex(
+        CompressionType.class, CompressionType::toString);
+
+    private final String value;
+
+    CompressionType(String value) {
+        this.value = value;
+    }
+
+    /**
+     * Maps the {@link CompressionType} to its corresponding {@link Compressor}.
+     * TODO: Update mappings here when additional compressors are supported in the future
+     */
+    public Compressor compressor() {
+        if (value == null) {
+            return null;
+        }
+        if (value.equals("gzip")) {
+            return new GzipCompressor();
+        }
+        return null;
+    }
```

Comment:
Can we move this to the `Compressor` interface instead, so it matches what we do in [SdkChecksum](https://github.com/aws/aws-sdk-java-v2/blob/305af519581f17ab2fc60a6dc4f5fb8b3c701e0f/core/sdk-core/src/main/java/software/amazon/awssdk/core/checksums/SdkChecksum.java#L52)

Reply:
good idea, migrating